.github/workflows/changelog.yml

```diff
@@ -0,0 +1,18 @@
+name: Check CHANGELOG.md
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  check-changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+      - name: Check if CHANGELOG.md has been modified
+        run: |
+          if git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | grep -q "CHANGELOG.md"; then
+            echo "CHANGELOG.md has been modified. Good job!"
+          else
+            echo "::error::CHANGELOG.md has not been modified. Please update the changelog."
```

Comment:
I think a warning is fine, no? Sometimes, we have small PRs (hotfixes or whatnot) that really don't need a changelog entry. It would be nice to not have to merge it with a red CI

Reply:
Indeed, let me see if I can make it a little bit less aggressive.

Reply:
@bobbinth @plafer 

<img width="1711" alt="SCR-20240716-rdzg" src="https://github.com/user-attachments/assets/b943d6c4-6b59-4b65-baeb-883e44d88a7b">

What do you think is the behaviour we should have here? 

It seems to me that the warning is too subtle, the job succeeds but throws a warning that most devs wouldn't see.

And plafer says that making the job fail is too aggressive, what do you think is the right thing to do?

Reply:
Yeah, I think that's too subtle, and my understanding is that there is no way to show a warning in the PR itself (only failure and success). If so, I think we should show a failure and then the admins can merge the PR anyway if in fact the PR doesn't need the changelog updated.

Reply:
Indeed, I was hoping for some "yellow CI" of some sort, but if this is the only way to do warnings, let's do errors instead